packages/core/strapi/lib/services/entity-service/types/params/populate.d.ts

```diff
@@ -0,0 +1,39 @@
+import type { Attribute, Common, Utils } from '@strapi/strapi';
+
+/**
+ * Wildcard notation for populate
+ *
+ * To populate all the root level relations
+ */
+type WildcardNotation = '*';
+
+/**
+ * Union of all possible string representation for populate
+ *
+ * @example
+ * type A = 'image'; // ✅
+ * type B = 'image,component'; // ✅
+ * type c = '*'; // ✅
+ * type D = 'populatableField'; // ✅
+ * type E = '<random_string>'; // ❌
+ */
+type StringNotation<TSchemaUID extends Common.UID.Schema> =
+  | WildcardNotation
+  | Attribute.GetPopulatableKeys<TSchemaUID>
```

Comment:
If you don't add a fallback to a regular string here, you won't be able to write

```ts
const populate: StringNotation<Common.UID.Schema> = 'foo';
```
 
 as foo is neither `'*'` nor `"${string},${string}"` or `never`

```suggestion
  | Utils.Guard.Never<Attribute.GetPopulatableKeys<TSchemaUID>, string>
```

Reply:
At the moment `Attribute.GetPopulatableKeys` returns a list of all relations, components, dynamic zones & media. However, do we consider morph relations (owner side) to be populated? (honestly not sure, might be better to test with the file.related attribute for instance)

Reply:
yeah but fallback incase of what ? you wouldn't want to populate something that doesn't exist

Reply:
Types not generated means any content type can have any field, so in this case, we should accept anything that can represent an attribute (=> a string)